doc/arkode/guide/source/Usage/LSRKStep/User_callable.rst

```diff
@@ -0,0 +1,319 @@
+
+   Specifies the safety factor for the Dominant Eigenvalues.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *domeigsfty* -- safety factor :math:`(>1)`.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument has an illegal value (e.g. ``domeigsfty < 1``)
+
+
+.. c:function:: int LSRKStepSetSSPStageNum(void* arkode_mem, int numofstages);
+
+   Sets the number of stages, ``s`` in ``SSP(s, p)`` methods.
+      
+      * ``ARKODE_LSRK_SSPs_2``  -- ``numofstages`` must be greater than or equal to 2
+      * ``ARKODE_LSRK_SSPs_3``  -- ``numofstages`` must be a full-square greater than or equal to 9
+      * ``ARKODE_LSRK_SSP10_4`` -- ``numofstages`` must be equal to 10 -- no need to call this function at all!
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *numofstages* -- number of stages :math:`(>1)` for ``SSP(s,2)`` and :math:`(n^2 = s \geq 9)` for ``SSP(s,3)``.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARKLS_MEM_NULL* ``arkode_mem`` was ``NULL``.
+      * *ARK_ILL_INPUT* if an argument has an illegal value (e.g. ``numofstages < 2``)      
+
+
+.. _ARKODE.Usage.LSRKStep.OptionalOutputs:
+
+Optional output functions
+------------------------------
+
+
+.. c:function:: int LSRKStepGetNumRhsEvals(void* arkode_mem, long int* fe_evals, long int* fi_evals);
+
+   Returns the number of calls to the user's right-hand
+   side functions, :math:`f^E` and :math:`f^I` (so far).
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *fe_evals* -- number of calls to the user's :math:`f^E(t,y)` function.
+      * *fi_evals* -- number of calls to the user's :math:`f^I(t,y)` function.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the ARKStep memory was ``NULL``
+
+
+.. c:function:: int LSRKStepGetNumDomEigUpdates(void* arkode_mem, long int* ndomeigupdates);
+
+   Returns the number of Dominant Eigenvalue evaluations (so far).
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *ndomeigupdates* -- number of calls to the user's ``DomEig`` function.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the ARKStep memory was ``NULL``     
+
+
+.. c:function:: int LSRKStepGetMaxStageNum(void* arkode_mem, int* stagemax);
+
+   Returns the max number of stages taken (so far).
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the LSRKStep memory block.
+      * *stagemax* -- max number of stages taken.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the ARKStep memory was ``NULL``  
+
+
+.. c:function:: int LSRKStepGetAverStageNum(void* arkode_mem, int* averstage);
```

Comment:
I would prefer that this be named either `LSRKStepGetAverageStageNum` or `LSRKStepGetAvgStageNum`, since "aver" is a non-standard abbreviation for "average".

Also, shouldn't this return a `sunrealtype` value, since averages are rarely integer-valued?